### PR TITLE
Add amqp and amqp external rules

### PR DESCRIPTION
**Pull request by Morganamilo**

### CODE REVIEW

File: docs/resources/rule_amqp.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "ably_rule_amqp Resource - terraform-provider-ably"
+subcategory: ""
+description: |-
+  The ably_rule_amql resource allows you to create and manage an Ably integration rule for AMQP. Read more at https://ably.com/docs/general/firehose/amqp-rule
```
Comment (graham-russell) [inline docs/resources/rule_amqp.md:6]: Should be ably_rule_amqp

File: docs/resources/rule_amqp.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+---
+
+# ably_rule_amqp (Resource)
+
+The `ably_rule_amql` resource allows you to create and manage an Ably integration rule for AMQP. Read more at https://ably.com/docs/general/firehose/amqp-rule
```
Comment (graham-russell) [inline docs/resources/rule_amqp.md:11]: Should be ably_rule_amqp

File: internal/provider/resource_ably_rule_amqp.go [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+				Type:        types.StringType,
+				Optional:    true,
+				Computed:    true,
+				Description: "JSON provides a text-based encoding",
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					DefaultAttribute(types.String{Value: "json"}),
+				},
+			},
+		},
+		"The `ably_rule_amql` resource allows you to create and manage an Ably integration rule for AMQP. Read more at https://ably.com/docs/general/firehose/amqp-rule"), nil
```
Comment (graham-russell) [inline internal/provider/resource_ably_rule_amqp.go:47]: Should be ably_rule_amqp

File: docs/resources/rule_amqp.md (hunk 1 above)
Comment (Morganamilo) [reply docs/resources/rule_amqp.md:6]: Whoops. Must have typod a search and replace.

**Review by graham-russell — APPROVED** [review]

**Review by kavalerov — APPROVED** [review]
LGTM, thanks!

**Pull request merged**